src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,41 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+
+        StringBuilder majorSB = new StringBuilder("Report for period");
+        String [] dateFromAR = dateFrom.split( " ");
+        String [] dateToAR = dateTo.split( " ");
+        String date1 = dateFromAR[dateFromAR.length -2].substring( 1);
+        String date2 = dateToAR[dateFromAR.length -2].substring(1,dateToAR[dateFromAR.length -2].length()-1);
```

Comment:
The logic for parsing `dateFrom` and `dateTo` seems incorrect. The task description specifies that the dates are in the format `dd.MM.yyyy`, but the current code attempts to parse them by splitting on spaces and using substring operations. This might not work as intended. Consider using `DateTimeFormatter` to parse the dates directly from the input strings.